src/conn/conn_api.c

```diff
@@ -1149,6 +1150,13 @@ __conn_reconfigure(WT_CONNECTION *wt_conn, const char *config)
 	__wt_free(session, conn->cfg);
 	conn->cfg = p;
 
+	/*
+	 * Fourth, run any diagnostic commands that were included. Do that last,
+	 * reconfiguration might include other configuration changes to support
+	 * a command.
+	 */
+	WT_ERR(__conn_verbose_cmd(session, config));
```

Comment:
> Would it make sense to move this into `__wt_verbose_config`

I thought about that myself, and didn't do it for two reasons: first, `__wt_verbose_config` doesn't take the `config` argument, and second, it means `__wt_verbose_config` has to be the last configuration function we call (so it's possible to configure stuff so a subsequent "diagnostic dump" command will work), and that seemed like a way to introduce a bug some day in the future. I don't feel strongly, let me know if you still prefer that change.

> and short-circuit there if verbose is the only option specified.

These APIs aren't likely to be called frequently: I think admins are more likely to commonly reconfigure more eviction worker threads, for example.

I think @michaelcahill went this way because "verbose" isn't an unreasonable name for this functionality, and implies MongoDB requires no additional plumbing.